### PR TITLE
Radix Sort

### PR DESCRIPTION
**Pull request by vipuluthaiah**
Implemented RadixSort in the dart.
Added in test cases

### CODE REVIEW

File: sort/radix.dart [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    maxDigits = Math.max(maxDigits, digitCount(number[i]));
+  }
+  return maxDigits;
+}
+
+radixSort(List nums) {
+  var maxDightCount = mostDigits(nums);
+  for (var k = 0; k < maxDightCount; k++) {
+    var digitBuckets = List.generate(10, (_) => []);
+    // print(digitBuckets);
```
Comment (Parowicz) [inline sort/radix.dart:41]: I guess it's debug print
```suggestion
```

File: sort/radix.dart [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  For elements with more than one significant digit, this bucketing process is repeated for each digit, 
+ while preserving the ordering of the prior step, until all digits have been considered. For this reason, 
+ radix sort has also been called bucket sort and digital sort.
+ */
+import 'dart:math' as Math;
+
+main() {
+  // print(radixSort([]));
+  var radixsort = radixSort([34, 2, 122, 24435, 23, 434, 232, 1323]);
+  print(radixsort);
```
Comment (Parowicz) [inline sort/radix.dart:13]:
```suggestion
  test("Sorting of empty list returns empty list", () {
    expect(radixSort([]), equals([]));
  });

  test("Sorting one element list return same list", () {
    expect(radixSort([1]), equals([1]));
  });

  test("Sorting two times doesnt change input", () {
    List<int> lst = [5, 7, 1, 10, 54, -1];
    expect(radixSort(lst), equals(radixSort(radixSort(lst))));
  });

  test("Sorting already sorted list returns same list", () {
    List<int> lst = [1, 2, 3, 4, 10];
    expect(radixSort(lst), equals(lst));
  });

  test("radix sort", () {
    expect(radixSort([34, -2, 122, 24435, 23, 434, 232, 1323]),
        equals([-2, 23, 34, 122, 232, 434, 1323, 24435]));
  });
```

File: sort/radix.dart [hunk 3]
```diff
@@ -0,0 +1,50 @@
+//radix sort
+/*
+ radix sort is a non-comparative sorting algorithm. It avoids comparison by creating and distributing elements into buckets according to their radix.
+  For elements with more than one significant digit, this bucketing process is repeated for each digit, 
+ while preserving the ordering of the prior step, until all digits have been considered. For this reason, 
+ radix sort has also been called bucket sort and digital sort.
+ */
+import 'dart:math' as Math;
```
Comment (Parowicz) [inline sort/radix.dart:8]: We are already using test framework
```suggestion
import 'package:test/test.dart';
import 'dart:math' as Math;
```

**Review by Parowicz — CHANGES_REQUESTED** [review]

File: sort/radix.dart (hunk 1 above)
Comment (vipuluthaiah) [reply sort/radix.dart:41]: yeahh! u want me to remove it?

File: sort/radix.dart (hunk 3 above)
Comment (vipuluthaiah) [reply sort/radix.dart:8]: what changes i should make now?

File: sort/radix.dart (hunk 1 above)
Comment (Parowicz) [reply sort/radix.dart:41]: Yea

File: sort/radix.dart (hunk 3 above)
Comment (Parowicz) [reply sort/radix.dart:8]: You can accept my suggestions or write your own test cases.

**Pull request closed without merging**